### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by Moro69**

### CODE REVIEW

File: camel-integration/src/main/java/com/moro/camel/integration/factory/EndpointFactory.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        }
+        OrderProcess orderProcess = new OrderProcess(orderWsdlURL);
+        return orderProcess.getOrderProcessPort();
+    }
+
+
+    public static ReviewEndpoint createReviewEndpoint() {
+        URL reviewWsdlURL = null;
+        try {
+            reviewWsdlURL = new URL("http://localhost:8088/beerfactory/beers/review?wsdl");
```
Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/factory/EndpointFactory.java]: string property

File: camel-integration/src/main/java/com/moro/camel/integration/factory/EndpointFactory.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
```
Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/factory/EndpointFactory.java]: what will happen next after catching exception?

File: camel-integration/src/main/java/com/moro/camel/integration/mbeans/Direction.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    public Direction() {
+        direction = "soap";
+    }
+
+    @Override
+    public void setDirection(String direction) {
+        this.direction = direction;
+        System.out.println("DIRECTION WAS SET TO " + direction);
```
Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/mbeans/Direction.java]: logging

File: camel-integration/src/main/java/com/moro/camel/integration/processor/SoapExecutorProcessor.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    private static final String BEERS_PATH = "/beerfactory/beers";
+    private static final String ORDERS_PATH = "/beerfactory/orders";
+
+    private ObjectMapper objectMapper;
+    private ModelMapper modelMapper;
+
+    @Autowired
+    public SoapExecutorProcessor(OrderSoapManager orderManager,
+                                    BeerSoapManager beerManager,
```
Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/processor/SoapExecutorProcessor.java]: formatting

File: camel-integration/src/main/java/com/moro/camel/integration/processor/SoapExecutorProcessor.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        URI uri = new URI(request.getRequestURI());
+        String uriPath = uri.getPath();
+        String method = request.getMethod();
+
+        LOGGER.info("URI IS " + uri);
+        LOGGER.info("URL IS " + request.getRequestURL());
+        LOGGER.info("METHOD IS " + method);
+        LOGGER.info("QUERY IS " + request.getQueryString());
+
+        if (uri.getPath().contains("review")) {
```
Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/processor/SoapExecutorProcessor.java]: terrible if statements

File: camel-integration/src/main/java/com/moro/camel/integration/rest/CamelRestRoute.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    private SoapExecutorProcessor soapExecutor;
+
+    @Autowired
+    public CamelRestRoute(SoapExecutorProcessor soapExecutor)
+            throws MalformedObjectNameException,
+            NotCompliantMBeanException,
+            InstanceAlreadyExistsException,
+            MBeanException {
+
+        direction = new Direction();
```
Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/rest/CamelRestRoute.java]: extract mbean to spring bean

File: camel-integration/src/main/java/com/moro/camel/integration/rest/CamelRestRoute.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+
+    @Override
+    public void configure() throws Exception {
+            from("jetty:http://localhost:8090?matchOnUriPrefix=true&bridgeEndpoint=true")
+                    .setProperty("direction", method(direction, "getDirection"))
+                    .choice()
+                        .when(exchangeProperty("direction").isEqualToIgnoreCase("rest"))
+                            .log("REDIRECT TO THE REST")
+                                .to("jetty:http://localhost:8085?bridgeEndpoint=true&throwExceptionOnFailure=false")
```
Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/rest/CamelRestRoute.java]: formatting

File: dao/src/main/java/com/moro/dao/implementation/OrderDaoImpl.java [hunk 8]
```diff
@@ -54,7 +55,7 @@ public OrderDaoImpl(NamedParameterJdbcTemplate namedParameterJdbcTemplate) {
     }
 
     @Override
-    public Collection<Order> getOrdersByDate(LocalDate fromDate, LocalDate toDate) {
+    public Collection<Order> getOrdersByDate(String fromDate, String toDate) {
```
Comment (ieiunium) [inline dao/src/main/java/com/moro/dao/implementation/OrderDaoImpl.java]: justify using String to store date instead LocalDate?

File: rest-service/src/main/java/com/moro/restService/BeerRestServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+        Collection<BeerDto> beers =
+                (Collection<BeerDto>) responseEntity.getBody();
+
+        return beers;
+    }
+
+    @Override
+    public Beer getBeerById(Integer beerId) {
+        Beer beer = restTemplate.getForObject(BEERS_URL + "/" + beerId, Beer.class);
```
Comment (ieiunium) [inline rest-service/src/main/java/com/moro/restService/BeerRestServiceImpl.java]: investigate how make rest template for sending request params

File: soap-app/src/main/java/com/moro/soap/implementation/BeerEndpointImpl.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+import com.moro.soap.interfaces.BeerEndpoint;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.jws.WebService;
+import java.util.ArrayList;
+
+@WebService(endpointInterface = "com.moro.soap.interfaces.BeerEndpoint",
+                serviceName = "BeerProcess",
```
Comment (ieiunium) [inline soap-app/src/main/java/com/moro/soap/implementation/BeerEndpointImpl.java]: formatting

File: camel-integration/src/main/java/com/moro/camel/integration/factory/EndpointFactory.java (hunk 2 above)
Comment (Moro69) [reply camel-integration/src/main/java/com/moro/camel/integration/factory/EndpointFactory.java]: we will see stack trace of exception and cannot create endpoints

File: rest-service/src/main/java/com/moro/restService/BeerRestServiceImpl.java (hunk 9 above)
Comment (Moro69) [reply rest-service/src/main/java/com/moro/restService/BeerRestServiceImpl.java]: i send request params when i want to get orders by specific date. should i send id as request param when need to get object by id or it's better to use path variable

File: dao/src/main/java/com/moro/dao/implementation/OrderDaoImpl.java (hunk 8 above)
Comment (Moro69) [reply dao/src/main/java/com/moro/dao/implementation/OrderDaoImpl.java]: it's easy to send soap requests using string to store date

File: camel-integration/src/main/java/com/moro/camel/integration/factory/EndpointFactory.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    private String ordersWsdl;
+    @Value("${review.wsdl}")
+    private String reviewWsdl;
+
+    public BeerEndpoint createBeerEndpoint() {
+        URL beerWsdlURL = null;
+        try {
+            beerWsdlURL = new URL(beersWsdl);
+        } catch (MalformedURLException e) {
+            LOGGER.error("Invalid url of beers wsdl. Check if it's available.", e);
```
Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/factory/EndpointFactory.java]: throw RuntimeException

File: camel-integration/src/main/java/com/moro/camel/integration/factory/EndpointFactory.java [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    private String beersWsdl;
+    @Value("${orders.wsdl}")
+    private String ordersWsdl;
+    @Value("${review.wsdl}")
+    private String reviewWsdl;
+
+    public BeerEndpoint createBeerEndpoint() {
+        URL beerWsdlURL = null;
+        try {
+            beerWsdlURL = new URL(beersWsdl);
```
Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/factory/EndpointFactory.java]: extract method wich creates URL, extract method wich creates URL

File: camel-integration/src/main/java/com/moro/camel/integration/processor/SoapExecutorProcessor.java [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+
+        HttpServletRequest request = exchange.getIn().getBody(HttpServletRequest.class);
+        URI uri = new URI(request.getRequestURI());
+        String uriPath = uri.getPath();
+        String method = request.getMethod();
+
+        beerByIdUriTemplate = new UriTemplate(beerByIdUri);
```
Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/processor/SoapExecutorProcessor.java]: is that thread-safe?

Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/processor/SoapExecutorProcessor.java]: is it correct initialization?

File: camel-integration/src/main/java/com/moro/camel/integration/processor/SoapExecutorProcessor.java [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+            LOGGER.info("ENDPOINT IS ORDERS");
+            LOGGER.info("OPERATION is addOrder");
+            orderManager.addOrder(exchange, objectMapper, modelMapper);
+        } else if(reviewUri.equals(uri)) {
+            LOGGER.info("ENDPOINT IS REVIEW");
+            LOGGER.info("OPERATION is addReview");
+            reviewManager.addReview(exchange, objectMapper, modelMapper);
+        } else throw new IllegalArgumentException("Invalid URI");
+    }
+    private void deleteMethodParser(Exchange exchange, String uri) {
```
Comment (ieiunium) [inline camel-integration/src/main/java/com/moro/camel/integration/processor/SoapExecutorProcessor.java]: add empty string

File: dao/src/main/java/com/moro/dao/implementation/OrderDaoImpl.java (hunk 8 above)
Comment (ieiunium) [reply dao/src/main/java/com/moro/dao/implementation/OrderDaoImpl.java]: what would you do if using localdate instead string will be more convenient for some other new dao client?

File: dao/src/main/java/com/moro/dao/implementation/OrderDaoImpl.java [hunk 15]
```diff
@@ -94,6 +95,10 @@ public Order addOrder(Order order) {
 
         KeyHolder keyHolder = new GeneratedKeyHolder();
 
+        if(order.getOrderDate() == null || order.getOrderDate().length() < 11) {
```
Comment (ieiunium) [inline dao/src/main/java/com/moro/dao/implementation/OrderDaoImpl.java]: 11 magic number?

File: dao/src/main/java/com/moro/dao/implementation/OrderDaoImpl.java (hunk 8 above)
Comment (Moro69) [reply dao/src/main/java/com/moro/dao/implementation/OrderDaoImpl.java]: it can be easy converted from string to localdate and vice versa

**Pull request merged**